src/test/aws/construct/static-asset-deployment.test.ts

```diff
@@ -24,89 +24,135 @@ const testStackProps = {
   stageContextPath: 'src/test/aws/common/cdkEnv',
 }
 
-class TestCommonStack extends CommonStack {
-  declare props: TestStackProps
-
-  constructor(parent: cdk.App, name: string, props: cdk.StackProps) {
-    super(parent, name, props)
-
-    this.construct = new TestStaticAssetDeployment(this, testStackProps.name, this.props)
-  }
+// get props
+function createTestStack<P extends CommonStackProps, T extends CommonConstruct<P>>(
+  app: App,
+  Construct: new (parent: Construct, id: string, props: P) => T,
+  id: string,
+  appProps: P
+): CommonStack {
+  class InitTestCommonStack extends CommonStack {
+    constructor(parent: cdk.App, id: string, props: P) {
+      super(parent, id, props)
 
-  protected determineConstructProps(props: cdk.StackProps) {
-    return {
-      ...super.determineConstructProps(props),
-      staticAssetBucket: this.node.tryGetContext('siteBucket'),
-      staticAssetDeployment: {
-        prune: true,
-        retainOnDelete: false,
-      },
-      staticAssetSources: [s3deploy.Source.asset('src/test/aws/common/resources/')],
-      staticAssetsForExport: [{ key: 'myCSV', value: 'test.csv' }],
+      this.construct = new Construct(this, id, { ...this.props, ...props })
+      // ideally this would be part of
+      if ('initResources' in this.construct && typeof this.construct.initResources === 'function') {
+        this.construct.initResources()
+      }
     }
   }
+
+  return new InitTestCommonStack(app, id, appProps)
 }
 
-class TestStaticAssetDeployment extends StaticAssetDeployment {
-  declare props: TestStackProps
+const createStack = (
```

Comment:
we could create a common function but for time being I would keep this as function for each test